src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -231,7 +266,139 @@ export const handleEncryptedSubmission: RequestHandler = async (
+      logger.error({
+        message: 'Unable to encrypt verified content',
+        meta: logMeta,
+        error,
+      })
+
+      // Passthrough if feature is not activated.
+      if (error instanceof MissingFeatureError) {
+        return next()
+      }
+
+      return res
+        .status(StatusCodes.BAD_REQUEST)
+        .json({ message: 'Invalid data was found. Please submit again.' })
+    }
+  }
+
+  // Verify Structure of Encrypted Response
+  // Step 1: Add req.body.encryptedContent to req.formData
+  // eslint-disable-next-line @typescript-eslint/no-extra-semi
+  ;(req as WithFormData<typeof req>).formData = req.body.encryptedContent
+  // Step 2: Add req.body.attachments to req.attachmentData
+  ;(req as WithAttachmentsData<typeof req>).attachmentData =
+    req.body.attachments || {}
+
+  // Save Responses to Database
+  const { formData, attachmentData } = req
+  const { verified } = res.locals
+  const attachmentMetadata = new Map()
+  const attachmentUploadPromises = []
+
+  // Object.keys(attachmentData[fieldId].encryptedFile) [ 'submissionPublicKey', 'nonce', 'binary' ]
+  for (const fieldId in attachmentData) {
+    const individualAttachment = JSON.stringify(attachmentData[fieldId])
+
+    const hashStr = crypto
+      .createHash('sha256')
+      .update(individualAttachment)
+      .digest('hex')
+
+    const uploadKey =
+      form._id + '/' + crypto.randomBytes(20).toString('hex') + '/' + hashStr
+
+    attachmentMetadata.set(fieldId, uploadKey)
+    attachmentUploadPromises.push(
+      AwsConfig.s3
+        .upload({
+          Bucket: AwsConfig.attachmentS3Bucket,
+          Key: uploadKey,
+          Body: Buffer.from(individualAttachment),
+        })
+        .promise(),
+    )
+  }
+
+  const submission = new EncryptSubmission({
+    form: form._id,
+    authType: form.authType,
+    myInfoFields: form.getUniqueMyInfoAttrs(),
+    encryptedContent: formData,
+    verifiedContent: verified,
+    attachmentMetadata,
+    version: req.body.version,
+  })
+
+  try {
+    await Promise.all(attachmentUploadPromises)
+  } catch (err) {
+    logger.error({
+      message: 'Attachment upload error',
+      meta: logMeta,
+      error: err,
+    })
+    return onEncryptSubmissionFailure(err, req, res, submission)
+  }
+
+  let savedSubmission
+  try {
+    savedSubmission = await submission.save()
+  } catch (err) {
+    return onEncryptSubmissionFailure(err, req, res, submission)
+  }
+
+  logger.info({
+    message: 'Saved submission to MongoDB',
+    meta: {
+      ...logMeta,
+      submissionId: savedSubmission._id,
+    },
+  })
+  req.submission = savedSubmission
+
+  // Fire webhooks if available
+  const webhookUrl = form.webhook?.url
+  const submissionWebhookView = submission.getWebhookView()
+  if (webhookUrl) {
+    // Note that we push data to webhook endpoints on a best effort basis
+    // As such, we should not await on these post requests
+    void pushData(webhookUrl, submissionWebhookView)
```

Comment:
can help to rename `pushData` to something more informative, like `webhook.send()`?

Reply:
I am renaming this as `webhookPushData` in this, maybe the rename can come later?